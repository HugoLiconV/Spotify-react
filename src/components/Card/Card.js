import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';

import cardStyle from '../../assets/styles/components/cardStyle';

function Card({ ...props }) {
  const { classes, className, children, plain, profile, ...rest } = props;
  const cardClasses = classNames({
    [classes.card]: true,
    [classes.cardPlain]: plain,
    [classes.cardProfile]: profile,
    [className]: className !== undefined
  });
  return (
    <div className={cardClasses} {...rest}>
      {children}
    </div>
  );
}

Card.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  plain: PropTypes.bool,
  profile: PropTypes.bool
};

export default withStyles(cardStyle)(Card);
