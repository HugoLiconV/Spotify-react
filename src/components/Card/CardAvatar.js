import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';

import cardAvatarStyle from '../../assets/styles/components/cardAvatarStyle';

function CardAvatar({ ...props }) {
  const { classes, children, className, plain, profile, ...rest } = props;

  const cardAvatarClasses = classNames(
    {
      [classes.cardAvatar]: true,
      [classes.cardAvatarProfile]: profile,
      [classes.cardAvatarPlain]: plain
    },
    className
  );
  return (
    <div className={cardAvatarClasses} {...rest}>
      {children}
    </div>
  );
}

CardAvatar.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  profile: PropTypes.bool,
  plain: PropTypes.bool
};

export default withStyles(cardAvatarStyle)(CardAvatar);
