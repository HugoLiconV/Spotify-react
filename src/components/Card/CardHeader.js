import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import cardHeaderStyle from '../../assets/styles/components/cardHeaderStyle';

function CardHeader({ ...props }) {
  const {
    classes,
    className,
    children,
    color,
    plain,
    stats,
    icon,
    ...rest
  } = props;
  const cardHeaderClasses = classNames(
    {
      [classes.cardHeader]: true,
      [classes[color + 'CardHeader']]: color,
      [classes.cardHeaderPlain]: plain,
      [classes.cardHeaderStats]: stats,
      [classes.cardHeaderIcon]: icon
    },
    className
  );
  return (
    <div className={cardHeaderClasses} {...rest}>
      {children}
    </div>
  );
}

CardHeader.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  color: PropTypes.oneOf(['info']),
  plain: PropTypes.bool,
  stats: PropTypes.bool,
  icon: PropTypes.bool
};

export default withStyles(cardHeaderStyle)(CardHeader);
