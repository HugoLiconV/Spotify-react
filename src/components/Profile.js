import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getCurrentUser } from '../actions/userActions';
class Profile extends Component {
  state = {
    user: {}
  };

  componentDidMount() {
    this.props.getCurrentUser();
  }

  render() {
    const user = this.props.currentUser;
    return (
      <div>
        <h1>Profile</h1>
        {console.log(user)}
      </div>
    );
  }
}

Profile.propTypes = {
  currentUser: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    currentUser: state.user.curretUser
  };
};

export default connect(
  mapStateToProps,
  { getCurrentUser }
)(Profile);
