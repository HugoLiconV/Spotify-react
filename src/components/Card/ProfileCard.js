import React from 'react';
import Button from '../Button/Button';
import Card from './Card';
import CardAvatar from './CardAvatar';
import CardBody from './CardBody';
import { withStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import noImageFound from '../../assets/img/no_image_found.png';

const styles = {
  details: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    '& *': {
      margin: '10px'
    }
  }
};

const currentUser = {
  country: 'MX',
  display_name: 'Hugo Licon',
  external_urls: {
    spotify: 'https://open.spotify.com/user/hugofernandoliconvalenzuela'
  },
  followers: {
    href: null,
    total: 12
  },
  images: [
    {
      height: null,
      url:
        'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5',
      width: null
    }
  ],
  product: 'premium',
  id: 'hugofernandoliconvalenzuela'
};

const getProfilePicture = currentUser => {
  if (currentUser.images.length === 0) return noImageFound;
  return currentUser.images[0].url;
};
export const ProfileCard = props => {
  const { currentUser, classes } = props;
  return (
    <Card profile>
      <CardAvatar profile>
        <a href={currentUser.external_urls.spotify}>
          <img
            style={{ backgroundColor: 'white' }}
            src={getProfilePicture(currentUser)}
            alt={currentUser.display_name}
          />
        </a>
      </CardAvatar>
      <CardBody profile>
        <div className={classes.details}>
          <h6 className={classes.cardCategory} data-product>
            Product: {currentUser.product}
          </h6>
          <h6 className={classes.cardCategory} data-country>
            Country: {currentUser.country}
          </h6>
          <h6 className={classes.cardCategory} data-followers>
            Followers: {currentUser.followers.total}
          </h6>
        </div>
        <h4 className={classes.cardTitle} data-name>
          {currentUser.display_name}
        </h4>
        <Button color="primary" round href={currentUser.external_urls.spotify}>
          Account
        </Button>
      </CardBody>
    </Card>
  );
};

ProfileCard.propTypes = {
  classes: PropTypes.object.isRequired,
  currentUser: PropTypes.object.isRequired
};

ProfileCard.defaultProps = {
  currentUser
};
export default withStyles(styles)(ProfileCard);
