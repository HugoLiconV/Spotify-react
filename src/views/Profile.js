import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getCurrentUser } from '../actions/userActions';
import GridContainer from '../components/Grid/GridContainer';
import GridItem from '../components/Grid/GridItem';
import ProfileCard from '../components/Card/ProfileCard';

class Profile extends Component {
  componentDidMount() {
    this.props.getCurrentUser();
  }

  render() {
    const { currentUser } = this.props;
    const isEmpty = Object.keys(currentUser).length === 0;
    if (!isEmpty) {
      return (
        <GridContainer justify="center">
          <GridItem xs={12} sm={12} md={6}>
            <ProfileCard currentUser={currentUser} />
          </GridItem>
        </GridContainer>
      );
    } else {
      return <h1>No current user</h1>;
    }
  }
}

Profile.propTypes = {
  currentUser: PropTypes.object.isRequired
};

Profile.defaultProps = {
  currentUser: {
    country: 'MX',
    display_name: 'Hugo Licon',
    external_urls: {
      spotify: 'https://open.spotify.com/user/hugofernandoliconvalenzuela'
    },
    followers: {
      href: null,
      total: 12
    },
    images: [
      {
        height: null,
        url:
          'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5',
        width: null
      }
    ],
    product: 'premium',
    id: 'hugofernandoliconvalenzuela'
  }
};

const mapStateToProps = state => {
  return {
    currentUser: state.user.curretUser
  };
};

export default connect(
  mapStateToProps,
  { getCurrentUser }
)(Profile);
