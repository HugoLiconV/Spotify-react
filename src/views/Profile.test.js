import React from 'react';
import { Profile } from './Profile';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ProfileCard } from '../components/Card/ProfileCard';

describe('<Profile /> component', () => {
  const setup = newProps => {
    const props = {
      getCurrentUser: jest.fn(),
      currentUser: {},
      ...newProps
    };

    const wrapper = shallow(
      <Profile
        currentUser={props.currentUser}
        getCurrentUser={props.getCurrentUser}
      />
    );

    return { wrapper, props };
  };

  it('should fetch currentUser', () => {
    const spy = jest.spyOn(Profile.prototype, 'componentDidMount');
    const { props } = setup();
    expect(spy).toHaveBeenCalled();
    expect(props.getCurrentUser).toHaveBeenCalled();
  });

  it('should show No current user if current user is empty', () => {
    const { wrapper } = setup();
    expect(wrapper.text()).toBe('No current user');
  });

  it('should show ProfileCard if current user is not empty', () => {
    const currentUser = {
      country: 'MX',
      display_name: 'Hugo Licon',
      external_urls: {
        spotify: 'https://open.spotify.com/user/hugofernandoliconvalenzuela'
      },
      followers: {
        href: null,
        total: 12
      },
      images: [
        {
          height: null,
          url:
            'https://scontent.xx.fbcdn.net/v/t1.0-1/p200x200/11880385_912284995474857_7569056323770464976_n.jpg?_nc_cat=109&_nc_ht=scontent.xx&oh=8dea37fe4bd092920aa0a9277f85d1f0&oe=5CD0F3C5',
          width: null
        }
      ],
      product: 'premium',
      id: 'hugofernandoliconvalenzuela'
    };
    const { wrapper } = setup({ currentUser });
    expect(wrapper.find('[data-profile-card]')).toExist();
  });

  it('should match snapshot', () => {
    const { wrapper: tree } = setup();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
