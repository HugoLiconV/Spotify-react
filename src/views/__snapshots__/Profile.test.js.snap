// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> component should match snapshot 1`] = `
<h1>
  No current user
</h1>
`;
